flang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1687,6 +1700,7 @@ void CompilerInvocation::setLoweringOptions() {
   const Fortran::common::LangOptions &langOptions = getLangOpts();
   loweringOpts.setIntegerWrapAround(langOptions.getSignedOverflowBehavior() ==
                                     Fortran::common::LangOptions::SOB_Defined);
+  loweringOpts.setStackArrays(codegenOpts.StackArrays);
```

Comment:
Not for this PR but it doesn't look right to me that stack arrays is in both lowering and codegen opts

Reply:
I added `StackArrays`, but then switched to `StackRepackArrays`. `StackArrays` is now unused, but I decided to keep it, as I want to "clean-up" our `-fstack-arrays` some time in future.  I am thinking about generating proper stack/heap allocations from the beginning rather than use the post-passes that may miss some cases. For that I am thinking about using `StackArrays` setting in both Lowering and further passes (like HLFIR-to-FIR).  I guess this still does not justify having the flag in two different sets.

I will just remove `StackArrays` from the lowering opts for now to avoid any confusion.